sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1749,6 +1749,19 @@ object SQLConf {
     .checkValue(v => v > 0, "The min partition number must be a positive integer.")
     .createOptional
 
+  val FILES_MAX_DESIRED_PARTITION_NUM = buildConf("spark.sql.files.maxDesiredPartitionNum")
+    .doc("The maximum desired number of partitions when reading files. When the number of " +
+      "partitions calculated for the first time is greater than this value, recalculate " +
+      s"${FILES_MAX_PARTITION_BYTES.key} so that the final number of partitions is close to this " +
+      "value. Note that the final calculated number of partitions may be larger than this value." +
```

Comment:
nit: a space at the end

Reply:
Since the doc is exposed to the end users, I think we should just emphasize the priority between this one and `spark.sql.files.maxPartitionBytes` instead of exposing the internal implementation algorithm. To avoid confusing user, maybe we should also update the doc of `spark.sql.files.maxPartitionBytes` to mention it is not guaranteed when this one is set.

Reply:
So, it seems the behavior of  `spark.sql.files.maxPartitionBytes` changed.

Reply:
No. It does not change `spark.sql.files.maxPartitionBytes`.